python/pyarrow/types.pxi

```diff
@@ -968,27 +988,6 @@ cdef class ExtensionType(BaseExtensionType):
         """
         return NotImplementedError
 
-    def __arrow_ext_class__(self):
```

Comment:
I am wondering if we would like to keep this here as well. It would give some duplication, but for someone looking at `ExtensionType` for implementing a subclass yourself, it's nice that the 4 special dunder methods are all together, and not spread between this class and the base class. 

(to avoid duplication, we can maybe leave out the docstring on the base class?)

Reply:
Moved `__arrow_ext_class__` and `__arrow_ext_scalar__` dunder methods to the `ExtensionType` class and left minimally documented implementations on `BaseExtensionType`.